### PR TITLE
Add buffer size property

### LINKED ISSUE #5: Remove hard limit on request size
As #4 points out, the request size can't grow.

### PR DESCRIPTION
**Pull request by karlfl**
Add a property with getter/setter to the HTTPServer class to allow adjustment of the incoming buffer size.   Developers can use this to read or change the size of the buffer used to store the incoming request data.  This is especially useful when the incoming request is larger than the default buffer size.  Without this change any incoming data over the default buffer size (1024 bytes) would be truncated.

Also made some adjustments to the documentation.  Adding sections for the temperature and polling examples, fixing an auto generated link and add helpful comment to the polling example.

This PR will close #5

### CODE REVIEW

File: adafruit_httpserver.py [hunk 1]
```diff
@@ -361,3 +361,25 @@ def poll(self):
                 # connection reset by peer, try again later.
                 return
             raise
+
+    @property
+    def requestbuffersize(self) -> int:
```
Comment (tekktrik) [inline adafruit_httpserver.py:366]: Snake case will make this a little more readable:

```suggestion
    def request_buffer_size(self) -> int:
```

Same thing for the setter of course!

File: examples/httpserver_simplepolling.py [hunk 2]
```diff
@@ -30,6 +30,7 @@ def base(request):  # pylint: disable=unused-argument
 
 while True:
     try:
+        # do something useful in this section
```
Comment (tekktrik) [inline examples/httpserver_simplepolling.py:33]: It make be helpful to give an example of what you mean by "something useful", like "operate hardware" or something.

**Review by tekktrik — CHANGES_REQUESTED** [review]
A couple suggestions, but maybe @dhalbert may know more about the effects of changing the request buffer size specifically.  Any reason this would be a problem?

File: examples/httpserver_simplepolling.py (hunk 2 above)
Comment (karlfl) [reply examples/httpserver_simplepolling.py:33]: Should be easy to add some more details to that comment.

File: adafruit_httpserver.py (hunk 1 above)
Comment (karlfl) [reply adafruit_httpserver.py:366]: I like the snake case and it's definitely easier to read.  I wasn't sure if there were naming standards, so I just mirrored the other function names.  I come from a camel case background, but that doesn't seem to apply in python except in class names.

Comment (tekktrik) [reply adafruit_httpserver.py:366]: Yup!  You've got it spot on.  I don't think I've ever seen it explicitly about the project, but generally we follow the [PEP8 style guide](https://peps.python.org/pep-0008/) (unless it conflicts with existing CPython API, like `logging`).

**Review by tekktrik — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**